tt_metal/common/tt_backend_api_types.hpp

```diff
@@ -169,3 +171,34 @@ template <>
 struct std::hash<tt::DataFormat> {
     std::size_t operator()(tt::DataFormat const& obj) const noexcept { return static_cast<std::size_t>(obj); }
 };
+
+template <>
+struct fmt::formatter<tt::DataFormat> : formatter<string_view> {
+    auto format(tt::DataFormat df, format_context& ctx) const -> format_context::iterator {
+        string_view name = "unknown";
+        switch (df) {
+            case tt::DataFormat::Bfp2: name = "Bfp2"; break;
+            case tt::DataFormat::Bfp2_b: name = "Bfp2_b"; break;
+            case tt::DataFormat::Bfp4: name = "Bfp4"; break;
+            case tt::DataFormat::Bfp4_b: name = "Bfp4_b"; break;
+            case tt::DataFormat::Bfp8: name = "Bfp8"; break;
+            case tt::DataFormat::Bfp8_b: name = "Bfp8_b"; break;
+            case tt::DataFormat::Float16: name = "Float16"; break;
+            case tt::DataFormat::Float16_b: name = "Float16_b"; break;
+            case tt::DataFormat::Float32: name = "Float32"; break;
+            case tt::DataFormat::Tf32: name = "Tf32"; break;
+            case tt::DataFormat::Int8: name = "Int8"; break;
+            case tt::DataFormat::UInt8: name = "UInt8"; break;
+            case tt::DataFormat::Lf8: name = "Lf8"; break;
+            case tt::DataFormat::UInt16: name = "UInt16"; break;
+            case tt::DataFormat::UInt32: name = "UInt32"; break;
+            case tt::DataFormat::Int32: name = "Int32"; break;
+            case tt::DataFormat::RawUInt8: name = "RawUInt8"; break;
+            case tt::DataFormat::RawUInt16: name = "RawUInt16"; break;
+            case tt::DataFormat::RawUInt32: name = "RawUInt32"; break;
+            case tt::DataFormat::Invalid: name = "Invalid"; break;
+            default: throw std::invalid_argument("Unknown format");
+        }
+        return formatter<string_view>::format(name, ctx);
```

Comment:
```suggestion
        const auto name = magic_enum::enum_name(df);
        
        if (not name) {
            throw std::invalid_argument("Unknown format");
        }
        
        return formatter<string_view>::format(*name, ctx);
```

Reply:
Right, I was reluctant to use the `magic_enum` in a header file. But I just realized that there's a corresponding cpp file, so I'll just move the implementation to it.